Feature/M0090IDE-3865_javafx_documentation (#388)
MicroEJ/docs: PlatformDeveloperGuide/mock.rst

- by adding the mock module as a regular Platform :ref:`module dependency <mmm_module_dependencies>` (if your Platform configuration project contains a ``module.ivy`` file), 
- or by manually copying the JAR file ``[mock_project]\target~\rip\mocks\[mock_name].jar`` to the :ref:`Platform configuration <platform_configuration_creation>` mock dropins folder ``dropins/mocks/dropins/``.

Use
===

Once installed, a Mock is used automatically by the Simulator when the
MicroEJ Application calls a native method which is implemented into the
Mock.

JavaFX
=======

> `JavaFX <https://openjfx.io/>`_ is a platform for creating Java applications with a modern user interface that is highly portable. It can be used to quickly create graphical Mocks for your Platform.

Reviewer: I suggest to remove application & platform terms to avoid confusion

```suggestion
`JavaFX <https://openjfx.io/>`_ is an open-source library for creating modern Java user interfaces that is highly portable. It can be used to quickly create graphical Mocks for your Platform.
```

> - Until the version 1.8, the Oracle JDK contains JavaFX, so this version allows you to use it right now in your project.

Reviewer: ```suggestion
- If your SDK is running on JDK 8, the Oracle JDK contains JavaFX, so this version allows you to use it right now in your project.
```

> - To use JavaFX with the JDK 11, this additional dependency must be added in your Mock project as well as your Platform project :

Reviewer: We should clarify the dependency purpose:
- in Mock project, this is a compile-time dependency (the content is not included in the mock module)
- then it must be added to the Platform project if your platform contains at least one mock (the content will be added to the platform module)

am I right ?

> - Until the version 1.8, the Oracle JDK contains JavaFX, so this version allows you to use it right now in your project.
> 
> - To use JavaFX with the JDK 11, this additional dependency must be added in your Mock project as well as your Platform project :

Reviewer: ```suggestion
- Until the version 1.8, the Oracle JDK contains JavaFX, so this version allows you to use it right now in your project.

- If your SDK is running on JDK 11, JavaFX must be added as an additional dependency to your Mock and Platform project. For that, MicroEJ Corp. provides a ready-to-use packaged module for all supported OS versions.
```

::

>     <dependency org="com.microej.tool" name="javafx" rev="1.1.0" />

Reviewer 2: is it on central repo?

Reviewer 3: Not yet. We plan to reduce the size of the rip before publishing it to central repository (should be done in the coming days).

Reviewer: @denisbedin please move to developer repo instead

Reviewer: please ask @gharelMicroEJ for a release

- or by manually copying the JAR file ``[mock_project]\target~\rip\mocks\[mock_name].jar`` to the :ref:`Platform configuration <platform_configuration_creation>` mock dropins folder ``dropins/mocks/dropins/``.

JavaFX
=======

> `JavaFX <https://openjfx.io/>`_ is a platform for creating Java applications with a modern user interface that is highly portable. It can be used to quickly create mocks for your Platform.

Reviewer 3: ```suggestion
`JavaFX <https://openjfx.io/>`_ is a platform for creating Java applications with a modern user interface that is highly portable. It can be used to quickly create graphical mocks for your Platform.
```

> - Until the version 1.8, JavaFX is bundled in the Oracle JDK, so this version allows you to use it right now in your project.

Reviewer 3: ```suggestion
- Until the version 1.8, the Oracle JDK contains JavaFX, so this version allows you to use it right now in your project.
```

> - To use JavaFX with the JDK 11, the additional dependency must be added in your Mock project as well as your Platform project :

Reviewer 3: ```suggestion
- To use JavaFX with the JDK 11, this additional dependency must be added in your Mock project as well as your Platform project :
```

::

    <dependency org="com.microej.tool" name="javafx" rev="1.1.0" />

> 

Reviewer 3: I think we should move this chapter at the end, after the `Use` section, since it is optional.

Author: The JavaFX section has been moved and the suggested changes have been applied.

- or by manually copying the JAR file ``[mock_project]\target~\rip\mocks\[mock_name].jar`` to the :ref:`Platform configuration <platform_configuration_creation>` mock dropins folder ``dropins/mocks/dropins/``.

JavaFX Module
=============

> The JavaFX Module allows the use of `JavaFX <https://openjfx.io/>`_ in the creation of a Mock project or to embed JavaFX in a Platform.

Reviewer 3: We can add a link to the official JavaFX site on the first occurrence of the JavaFX word.

> The JavaFX Module allows the use of `JavaFX <https://openjfx.io/>`_ in the creation of a Mock project or to embed JavaFX in a Platform.

Author: The JavaFX section has been simplified. It was not needed to create two parts for each artifact (jar and rip).

This Module is required when working with a JDK 11, since JavaFX is not embedded anymore in Oracle JDK.

The following OS are supported by the JavaFX Module :

- Linux 64
- Mac Aarch64
- Mac x86_64
- Windows 64

> To ensure Mac Aarch64 support, the JavaFX Module is based on JavaFX 17 which requires a JDK 11, so it can not be used with a JDK 1.8.

Reviewer 3: Even without the support of the Mac M1, we cannot use it with a JDK 8 since there is no version for the JDK, no?

Author: Yes , this sentence has been removed
